reproducibility_project/lrc_shift_subproject/src/engines/gromacs/project.py

```diff
@@ -148,32 +163,135 @@ def gmx_nvt(job):
+@Project.post(lambda j: j.isfile("nvt_prod.gro"))
+@flow.with_job
+@flow.cmd
+def gmx_nvt_prod(job):
+    """Run GROMACS grompp for the nvt step."""
+    nvt_mdp_path = "nvt_prod.mdp"
+    grompp = f"gmx_mpi grompp -f {nvt_mdp_path} -o nvt_prod.tpr -c npt_prod.gro -p init.top --maxwarn 1"
+    mdrun = _mdrun_str("nvt_prod")
+    return f"{grompp} && {mdrun}"
+
+
+@Project.operation
+@Project.pre(lambda j: j.sp.engine == "gromacs")
+@Project.pre(lambda j: j.isfile("nvt_prod.gro"))
+@Project.pre(lambda j: not equil_status(j, "nvt_prod", "Potential"))
+@Project.pre(lambda j: not equil_status(j, "nvt_prod", "Pressure"))
+@Project.post(lambda j: equil_status(j, "nvt_prod", "Potential"))
+@Project.post(lambda j: equil_status(j, "nvt_prod", "Pressure"))
+@flow.with_job
+@flow.cmd
+def extend_gmx_nvt_prod(job):
+    """Run GROMACS grompp for the nvt step."""
+    # Extend the npt run by 1000 ps (1 ns)
+    extend = "gmx_mpi convert-tpr -s nvt_prod.tpr -extend 1000 -o nvt_prod.tpr"
+    mdrun = _mdrun_str("nvt_prod")
     return f"{extend} && {mdrun}"
 
 
+@Project.operation
+@Project.pre(lambda j: j.sp.engine == "gromacs")
+@Project.pre(lambda j: j.isfile("npt_prod.gro"))
+@Project.pre(lambda j: equil_status(j, "npt_prod", "Potential"))
+@Project.pre(lambda j: equil_status(j, "npt_prod", "Volume"))
+@flow.with_job
+def sample_npt_properties(job):
+    """Sample properties of interest from npt edr."""
+    import pandas as pd
+
+    from reproducibility_project.src.analysis.sampler import sample_job
+
+    p = pathlib.Path(job.workspace())
+    data = panedr.edr_to_df(f"{str(p.absolute())}/npt_prod.edr")
+    # Properties of interest
+    poi = {
+        "Time": "time",
+        "Potential": "potential_energy",
+        "Kinetic En.": "kinetic_energy",
+        "Pressure": "pressure",
+        "Temperature": "temperature",
+        "Density": "density",
+        "Volume": "volume",
+    }
+
+    tmp_df = pd.DataFrame()
+    for idx, row in data.iterrows():
+        tmp_df = tmp_df.append(row[list(poi.keys())])
+    tmp_df.rename(poi, axis=1, inplace=True)
+    tmp_df.insert(
+        column="time_steps",
+        value=[10000 * i for i in range(len(tmp_df))],
+        loc=1,
+    )
+
+
+@Project.operation
+@Project.pre(lambda j: j.sp.engine == "gromacs")
+@Project.pre(lambda j: j.isfile("nvt_prod.gro"))
+@Project.pre(lambda j: equil_status(j, "nvt_prod", "Potential"))
+@Project.pre(lambda j: equil_status(j, "nvt_prod", "Pressure"))
+@flow.with_job
+def sample_nvt_properties(job):
+    """Sample properties of interest from nvt edr."""
+    import pandas as pd
+
+    from reproducibility_project.src.analysis.sampler import sample_job
+
+    p = pathlib.Path(job.workspace())
+    data = panedr.edr_to_df(f"{str(p.absolute())}/nvt_prod.edr")
+    # Properties of interest
+    poi = {
+        "Time": "time",
+        "Potential": "potential_energy",
+        "Kinetic En.": "kinetic_energy",
+        "Pressure": "pressure",
```

Comment:
Pressure probably less important as a poi in the NVT sims, id include volume and density here